tests/ui/resolve/issue-109343.stderr

```diff
@@ -0,0 +1,28 @@
+error[E0463]: can't find crate for `f`
+  --> $DIR/issue-109343.rs:2:1
+   |
+LL | extern crate f;
+   | ^^^^^^^^^^^^^^^ can't find crate
+
+error[E0432]: unresolved import `inner`
+  --> $DIR/issue-109343.rs:5:9
+   |
+LL | pub use inner::f;
+   |         ^^^^^ maybe a missing crate `inner`?
+   |
+   = help: consider adding `extern crate inner` to use the `inner` crate
+
+error: extern crate `f` is private, and cannot be re-exported (error E0365), consider declaring with `pub`
+  --> $DIR/issue-109343.rs:9:9
+   |
+LL | pub use f as g;
+   |         ^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
```

Comment:
Is the issue specific to this case, by the way (public reexports of private `extern crate`s)?

The fact that it is allowed is a pretty bad hack that exists purely for backward compatibility, and it does put the resolver structures into an inconsistent state impossible under normal circumstances.

Reply:
> Is the issue specific to this case,

I'm not sure, but so far it looks like it

>  it does put the resolver structures into an inconsistent state impossible under normal circumstances.

Yes, I agree that there will be a problem, and it is not a good idea to change the existing architecture to solve a realistic scenario that will hardly ever be encountered.

Maybe we can ignore this issue for now, and wait until the architecture can solve this problem very well one day.